homeassistant/components/here_weather/config_flow.py

```diff
@@ -0,0 +1,120 @@
+"""Config flow for here_weather integration."""
+from __future__ import annotations
+
+import logging
+
+import herepy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+)
+from homeassistant.core import HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+
+from .const import DEFAULT_MODE, DEFAULT_SCAN_INTERVAL, DOMAIN, HERE_API_KEYS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_validate_user_input(hass: HomeAssistant, user_input: dict) -> None:
+    """Validate the user_input containing coordinates."""
+    here_client = herepy.DestinationWeatherApi(user_input[CONF_API_KEY])
+    await hass.async_add_executor_job(
+        here_client.weather_for_coordinates,
+        user_input[CONF_LATITUDE],
+        user_input[CONF_LONGITUDE],
+        herepy.WeatherProductType[DEFAULT_MODE],
+    )
+
+
+class HereWeatherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for here_weather."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
```

Comment:
This is legacy. We now set the iot class in the manifest.json.

https://developers.home-assistant.io/docs/creating_integration_manifest#iot-class

Reply:
Done in https://github.com/home-assistant/core/pull/28910/commits/8a0dd2325b18979a6f8ed0d77837555e52cfd384